test/validate-schemas.spec.ts

```diff
@@ -0,0 +1,51 @@
+import fs from 'fs-extra';
+import upath from 'upath';
+import { Json } from '../lib/util/schema-utils';
+import { capitalize } from '../tools/docs/utils';
+import * as Schemas from '../tools/schemas/schema';
+
+const dataFileDir = 'lib/data';
+const schemaDir = 'tools/schemas';
+const schemasAndJsonFiles: {
+  schemaName: keyof typeof Schemas;
+  dataFileName: string;
+}[] = [];
+
+describe('validate-schemas', () => {
+  it('validate json files in lib/data against their schemas', async () => {
+    const schemaFiles = (await fs.readdir(schemaDir)).filter(
+      (file) => upath.extname(file) === '.json',
+    );
+
+    for (const schemaFile of schemaFiles) {
+      const correspondingDatFileName = schemaFile.replace('-schema', '');
+      const schemaName = `${schemaFile
+        .replace('.json', '')
+        .split('-')
+        .map(capitalize)
+        .join('')}` as keyof typeof Schemas;
+      schemasAndJsonFiles.push({
+        schemaName,
+        dataFileName: correspondingDatFileName,
+      });
+    }
+
+    await Promise.all(
```

Comment:
use all settled, so we wait for all promises to settle, otherwise it will return early and we've lingering promises

Reply:
When using all settled the tests pass regardless of the assertions being false :)